### ISSUE TITLE
Conscript should be more flexible about launchconfig locations

### ISSUE THREAD

**Issue by clhodapp** [foundweekends/conscript#63]
Right now, conscript searches for launchconfig files using the pattern "^src/main/conscript/([^/]+)/launchconfig$". The problem with this is that it's too inflexible. I personally think that launchconfig files are not part of the source of my application and don't want them in src/main (or even in src). I expect that a lot of people _do_ think of their launchconfigs as part of their sources, but maybe they don't keep the source files for their application at the top of their repository in a directory called "src". For this reason, I'd advocate changing the pattern to something more like "(?:^|.+/)conscript/([^/]+)/launchconfig$". This pattern should match any files that are matched now (I think), but also allow e.g. a "conscript" directory at the top of the repository or one inside of a project that isn't at the top. Thoughts?

**Comment by eed3si9n** [comment]
I think this is a good idea. Does Github have an API that can query for this pattern?
If someone wants to implement this, I'd be happy to review a pull request.

**Comment by michalmela** [comment]
Indeed, it seems that Github provides an API for recursively listing all the objects (blobs or trees) in a particular tree. One of the HTTP requests that Conscript makes (for, e.g., giter8) is for this URL: 

https://api.github.com/repos/foundweekends/giter8/git/trees/9bfc23a17fe35f52e8b5f5ec348f0fd6a396d975?recursive=1

Of particular interest under the URL above is this object:

    {
      "path": "src/main/conscript/g8/launchconfig",
      "mode": "100644",
      "type": "blob",
      "sha": "58800cd1cf1f81e53600b5381acfec71039a9d45",
      "size": 231,
      "url": "https://api.github.com/repos/foundweekends/giter8/git/blobs/58800cd1cf1f81e53600b5381acfec71039a9d45"
    }

So one can see that the full path can be retrieved easily Therefore, simply replacing the Regex used for finding the `launchconfig` file should do the job. I can prepare the pull request.
